src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -803,14 +803,22 @@ public static bool CheckConstraints(
             this MethodSymbol method,
             in CheckConstraintsArgs args)
         {
-            if (!RequiresChecking(method))
+            Debug.Assert(!method.GetIsNewExtensionMember()); // Tracked by https://github.com/dotnet/roslyn/issues/76130: Review all remaining callers with regards to new extensions
```

Comment:
>Review all remaining callers with regards to new extensions [](http://example.com/codeflow?start=122&length=59)

Is the goal to remove the enclosing method or ```CheckConstraintsIncludingExtension``` at the end? If so, consider saying this explicitly. If not, could you please elaborate what is the goal?

Reply:
The question was whether we want any of the callers be affected by extensions. If not, they'll continue to use this method. Otherwise, their logic needs to be updated (lookup logic) and they should call `CheckConstraintsIncludingExtension`.
After reviewing the callers, I see that none of them accounts for classic extension methods, so I don't think they should be affected by new extension methods either.
Will remove this follow-up comment

Reply:
I am sorry, but the response doesn't make sense to me. We don't want the callers to be affected , but we call ```CheckConstraintsIncludingExtension``` here. So, why is it a good thing to have two methods after you reviewed all callers?

Reply:
Resolved offline. Brought back original name